### PR TITLE
mongocxx: 3.8.2 → 3.10.2

### PR DESCRIPTION
**Pull request by Vcele**
## Description of changes

- Updated mongocxx from 3.8.1 to 3.10.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Vcele** [comment]
@IogaMaster Thanks for your guide. Maybe you can take a look ;)

**Comment by jackyliu16** [comment]
Based on convention, you should change the title of this PR from ` Update mongocxx #324374 ` into `mongocxx: 3.8.2 → 3.10.2`

**Comment by Vcele** [comment]
> Based on convention, you should change the title of this PR from `Update mongocxx #324374` into `mongocxx: 3.8.2 → 3.10.2` This will make it easier for others to view the specific behavior of the PR implementation

Ok thank's

**Comment by jackyliu16** [comment]
Sorry, you should spilt the action of add maintainer and update package in two commit. It was my mistake.

**Comment by jackyliu16** [comment]
And Base on [Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) you could add following into your Description of changes.

`Closes #323974`

**Comment by Vcele** [comment]
I am just struggling a bit with git. I don't quite know how the interactive rebase editor works yet.

**Comment by Vcele** [comment]
Closes #323974

**Comment by Vcele** [comment]
> I am just struggling a bit with git. I don't quite know how the interactive rebase editor works yet.

Ok got It can you take a look if it's all correct now?

**Comment by jackyliu16** [comment]
The commit message is correct right now. Great jobs

**Review by adriandole — APPROVED** [review]

**Comment by Vcele** [comment]
```pkgs.darwin.apple_sdk.frameworks.Security```
How can I add this as an dependency only for darwin?

**Comment by jackyliu16** [comment]
> `Security`

Maybe you could use some kinds of things like 
 buildInputs = [] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;

**Comment by Vcele** [comment]
> @Vcele 
> 
> Maybe you could use some kinds of things like 
>  buildInputs = [] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
>  
> I not a kind of darwin user ( i not even have one ), so just a guess.

Me neither, but I can try that. I just saw that the build failed for that system in the checks.

**Comment by Vcele** [comment]
@adriandole if you've got time to give this another review maybe you got time to merge this via the [nixpkgs-merge-bot](https://github.com/NixOS/nixpkgs-merge-bot) 👍

**Comment by Vcele** [comment]
Result of `nixpkgs-review pr 324374` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mongocxx</li>
  </ul>
</details>

**Review by IogaMaster — APPROVED** [review]
Thanks for watching my video!
It looks good to me!

**Comment by Frontear** [comment]
Result of `nixpkgs-review pr 324374` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mongocxx</li>
  </ul>
</details>

File: pkgs/development/libraries/mongocxx/default.nix
Comment (Aleksanaa) [inline pkgs/development/libraries/mongocxx/default.nix:56]: You can override in top-level: `stdenv = darwin.apple_sdk_11_0.stdenv`

Comment (Vcele) [reply pkgs/development/libraries/mongocxx/default.nix:58]: Ok 👍

**Comment by tomberek** [comment]
@NixOS/nixpkgs-merge-bot merge

**Pull request merged**
